src/Rocketlet.ts

```diff
@@ -64,6 +65,17 @@ export abstract class Rocketlet {
     }
 
     /**
+     * This method defines in which relationship to Rocket.Chat this Rocketlet lives.
+     * This information is consumed within the visualization (e. g. for each associated entity,
+     * a different set of visualization options is applicable) and in the persistence.
+     *
+     * @return The entity referred to and the cardinality of this association (e. g. can a room
+     * extended by this Rocketlet have one or multiple Rocketlet data items?
+     *
+     */
+    public abstract getRocketChatAssociation(): IRocketChatAssociation;
```

Comment:
I am struggling to see the point of this, since a Rocketlet can extend more than one part of the Rocket.Chat system.

An example I keep going back to is a GitHub Rocketlet where it adds commands for interacting with GitHub items, adds a tab with different pieces related to the conversation and issues on GitHub, provides a way to comment on Github issues inside Rocket.Chat, provide action buttons for an individual when an issue is opened or they're assigned an issue, and various other things.

So, I'm not sure this association should happen and we should stick with the interfaces defining how this Rocketlet extends Rocket.Chat.

Unless I am completely misunderstanding the purpose of this.